cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -3279,6 +3299,195 @@ TEST(Substrait, IsthmusPlan) {
                        *compute::default_exec_context(), buf, {}, conversion_options);
 }
 
+TEST(Substrait, PlanWithExtension) {
```

Comment:
Can we change the name of this test to reflect that it has something to do with asof join?

Reply:
Will do.